ext/standard/array.c

```diff
@@ -3801,29 +3801,37 @@ PHP_FUNCTION(array_change_keys)
         fci.no_separation = 0;
 
         if (str_key) {
-            ZVAL_STR(&params[0], zend_string_copy(str_key));
+            ZVAL_STR_COPY(&params[0], str_key);
         } else {
             ZVAL_LONG(&params[0], num_key);
         }
         ZVAL_COPY(&params[1], value);
 
         if (zend_call_function(&fci, &fci_cache) != SUCCESS || Z_TYPE(result) == IS_UNDEF) {
             zval_dtor(return_value);
-            zval_ptr_dtor(&params[0]);
+            zval_dtor(&params[0]);
+            zval_dtor(&params[1]);
+            efree(params);
```

Comment:
Previously, `params` was never freed on error.

Also, `zval_dtor` seems to be the correct method here, as the params are effectively temporary variables. I researched the difference between `zval_dtor` and `zval_ptr_dtor` in the [PHP Internals Book](https://nikic.github.io/2012/03/28/Understanding-PHPs-internal-array-implementation.html):

> The difference between zval_ptr_dtor and zval_dtor is that zval_ptr_dtor only destroys the zval if the refcount reaches 0, whereas zval_dtor destroys it always, regardless of the refcount. That’s why you’ll find zval_ptr_dtor used with “normal” values and zval_dtor with temporary variables, which aren’t used anywhere else anyways. Also zval_ptr_dtor frees the zval after destroying it, while zval_dtor does not.

This also seems consistent with other `zend_call_function()` examples elsewhere in the array functions.